tests/profiling/test_accuracy.py

```diff
@@ -105,3 +105,201 @@ def test_accuracy_stack_v2():
+    durations: list[float] = []
+    for _ in range(15):
+        start = time.thread_time_ns()
+        cpu_loop(n)
+        durations.append(time.thread_time_ns() - start)
+    durations.sort()
+    # Trim 1 sample from each end to remove outliers (13 samples remaining)
+    trimmed = durations[1:-1]
+    median_ns = trimmed[len(trimmed) // 2]  # 7th sample (middle of 13)
+
+    # Calculate iterations per second based on median
+    iterations_per_second = n / (median_ns / 1_000_000_000)
+
+    # Calculate exact iterations needed for target time
+    target_iterations = cpu_seconds * iterations_per_second
+
+    # Verify the calibration by measuring the actual time
+    # If it's significantly off, adjust
+    start = time.thread_time_ns()
+    cpu_loop(int(target_iterations))
+    actual_time_ns = time.thread_time_ns() - start
+
+    # If actual time is more than 20% off, adjust proportionally
+    if actual_time_ns > 0:
+        ratio = cpu_nano / actual_time_ns
+        target_iterations = int(target_iterations * ratio)
+
+    return int(target_iterations)
+
+
+@pytest.mark.subprocess(
+    env=dict(
+        DD_PROFILING_OUTPUT_PPROF="/tmp/test_accuracy_cpu_loop.pprof",
+        _DD_PROFILING_STACK_V2_ADAPTIVE_SAMPLING_ENABLED="0",
+    )
+)
+def test_cpu_loop_accuracy():
+    import collections
+    import os
+    import time
+
+    from ddtrace.profiling import profiler
+    from tests.profiling.collector import pprof_utils
+    from tests.profiling.test_accuracy import assert_almost_equal
+    from tests.profiling.test_accuracy import calibrate_cpu_loop
+    from tests.profiling.test_accuracy import cpu_loop
+
+    # Start profiler for calibration (to account for profiler overhead)
+    p_cal = profiler.Profiler()
+    p_cal.start()
+
+    # Calibrate: find iterations needed for 0.01 seconds (10 milliseconds)
+    # This runs with profiler on to account for profiler overhead
+    iterations = calibrate_cpu_loop(0.01)
+
+    # Stop profiler and flush to clear the profile data from calibration
+    p_cal.stop(flush=True)
+
+    # Remove the pprof file created during calibration
+    pprof_file = os.environ["DD_PROFILING_OUTPUT_PPROF"] + "." + str(os.getpid())
+    if os.path.exists(pprof_file):
+        os.remove(pprof_file)
+
+    # Start profiler for the actual test
+    p = profiler.Profiler()
+    p.start()
+
+    # Run cpu_loop for 0.01 seconds, then sleep 0.1 seconds, repeat 400 times
+    # The number of iterations (400) was chosen to:
+    # - Generate enough samples for accurate measurement
+    # - Keep test duration under 1 minute threshold
+    # Total wall time: 400 * (0.01 + 0.1) = 44 seconds
+    # Total CPU time on cpu_loop: 400 * 0.01 = 4 seconds
+    # Measure actual CPU time spent to verify calibration
+    actual_cpu_time_ns = 0
+    for _ in range(400):
+        start_cpu = time.thread_time_ns()
+        cpu_loop(iterations)
+        actual_cpu_time_ns += time.thread_time_ns() - start_cpu
```

Comment:
This code is repeated 4-5 times, let's make it a helper that returns
`time.thread_time_ns() - start_cpu`